homeassistant/components/harman_kardon_avr/media_player.py

```diff
@@ -71,17 +71,28 @@ def __init__(self, avr):
         self._current_source = avr.current_source
 
     def update(self) -> None:
-        """Update the state of this media_player."""
-        if self._avr.is_on():
-            self._attr_state = MediaPlayerState.ON
-        elif self._avr.is_off():
+        """Update the state of this media_player by pinging the device."""
+        try:
+            # Send one ping packet with a 1-second timeout.
+            result = subprocess.run(
+                ["ping", "-c", "1", "-W", "1", self._host],
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL,
+            )
```

Comment:
This kinda of logic does not belong in Home Assistant, but in the library handling the connection (the abstraction). That said, pings like this, in general, is very fragile.